doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,149 @@
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional. However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
+ * the field of scalars ``F`` are the real numbers.
+
+It is possible to use the class without fulfilling these assumtptions, however in this case some of the mathematical results might not hold.
+
+
+Implementation of functionals
+=============================
+
+To define your own functional, you start by writing::
+
+    class MyFunctional(odl.solvers.Functional):
+        ...
+
+Since `Functional` is a subclass of `Operator`, it has the *abstract method* ``domain`` which need to be explicitly overridden by any subclass.
+
+``domain``: `Set`
+    The domain of this functional, i.e., the set of elements to which this functional can be applied.
+
+Moreover, there are several optional parameters associated with a functional. These are
+
+``linear`` : `bool`, optional
+    If `True`, the functional is considered as linear. In this case, ``domain`` and ``range`` have to be instances of `LinearSpace`, or `Field`.
+    Default: ``False``
+smooth : `bool`, optional
+    If `True`, assume that the functional is continuously differentiable.
+    Default: ``False``
+concave : `bool`, optional
+    If `True`, assume that the functional is concave.
+    Default: ``False``
+convex : `bool`, optional
+    If `True`, assume that the functional is convex.
+    Default: ``False``
+grad_lipschitz : 'float', optional
+    The Lipschitz constant of the gradient.
+    Default: infinity
+
+A functional also has a number of optional methods and properties associated with it. The default value of these are all to raise a `NotImplemetedError`. The properties are:
+
+ * ``gradient``. This returns the gradient operator of the functional, i.e., the operator that corresponds to the mapping
+
+.. math::
+
+    x \\to \\nabla f(x)
+
+where :math:`\\nabla f(x)` is the element used to evaluate derivatives in a direction :math:`d` by :math:`\\langle \\nabla f(x), d \\rangle`.
+ * ``conjugate_functional``. This is the convex conjugate functional, also known as the Legendre transform or Fenchel conjugate. It is defined as 
+
+.. math::
+
+    f^*(x^*) = \sup_{x \in X} \{ \langle x^*,x \rangle - f(x)  \}.
+
+For general linear spaces :math:`X`, :math:`x^* \in X^*` the (continuous/normed) dual space of :math:`X`, i.e., the space of all continuous linear functionals defined on :math:`X`. Then :math:`\langle x^*,x \rangle` is the "dual pairing", i.e., the evaluation of the linear functional :math:`x^*` in the point :math:`x`. However, Hilbert spaces are self-dual, meaning :math:`X^* = X`, and :math:`\langle x^*,x \rangle` is the inner product.
```

Comment:
Yes, you need to indicate by indention that everything belongs to the same logical unit, the list item. This means that the `.. math::` directive and the `For ...` text both need to have the same indentation as the text at the
```
``conjugate_functional``
```
text.
Also, try not to cram too much into one text line. This one line spans 5 lines in the online viewer.
A (in my view) good guideline from the combination of Git with LaTeX is to use one sentence per line, because text is usually edited sentence by sentence. Using one line per sentence best fits the desire to have readable diffs between Git versions. (I have myself not been following this principle so far, but I think it's best and am going to adhere to it.)